src/index.ts

```diff
@@ -168,7 +168,11 @@ class Compiler {
     switch (typeof a) {
       case 'boolean': return new TsType.Boolean // ts doesn't support literal boolean types
       case 'number': return new TsType.Number   // ts doesn't support literal numeric types
-      case 'string': return new TsType.Literal(JSON.stringify(a))
+      case 'string':
+        // output single quotes for tslint with inner quotes (if any) escapped
+        var stringValue = JSON.stringify(a)
+        stringValue = `'${stringValue.substring(1, stringValue.length - 1).replace(/'/g,'\\\'')}'`
+        return new TsType.Literal(stringValue)
```

Comment:
Tests are a good idea. I can do that. This is connected to some of the changes though.

For example:
* https://github.com/bcherny/json-schema-to-typescript/pull/36/commits/150110e18d5cccb201eeaa37b7668cba96f46544#diff-22946053a913fc1293f4df407a6992f4R44 changes the double quote to single quote. And the above code change in index.ts is required to validate successfully.
* There are many other locations of the double quotes too.

So I will break this up into 2 pull requests. One just for the double quote tslint issues and the other for all remaining tslint issues.